api/v1alpha1/gateway_types.go

```diff
@@ -238,6 +238,39 @@ type LocalObjectReference struct {
 	Name string `json:"name" protobuf:"bytes,3,opt,name=name"`
 }
 
+// RouteBindingSelector defines a schema for associating routes with the Gateway.
+// No routes will be associated to the Gateway if NamespaceSelector and RouteSelector
+// are undefined. If both NamespaceSelector and RouteSelector are defined, only
+// routes matching the RouteSelector within the NamespaceSelector namespaces are
+// associated with the Gateway.
```

Comment:
Reading all the doc comments, I'm concluding that if nothing is defined, then the behavior is same as Ingress v1, i.e., this gateway associates all the routes in all namespaces.

Can we explicitly mention this to avoid confusion down the road?

Reply:
@hbagdi I stuck with the previous defaulting behavior of `Routes`, meaning no routes are associated to a gateway if `NamespaceSelector` and `RouteSelector` are undefined. Per feedback from @bowei during last week's meeting, I'm having the behavior of these newly introduced properties match the behavior of [metav1.LabelSelector](https://godoc.org/github.com/ericchiang/k8s/apis/meta/v1#LabelSelector).